cli/azd/pkg/ext/hooks_runner.go

```diff
@@ -7,6 +7,7 @@ import (
 	"os"
 	"strings"
 
+	"github.com/azure/azure-dev/cli/azd/pkg/environment"
```

Comment:
I guess we can't save ourselves from creating this dependency here, right?  We want to call `Reload` inside `Invoke` ...

I'm curious..  if we move `Reload()` to be called from inside `Environ()`, would that give us the chance to keep `[]string` for env within `ext` package?  anyone calling `Environ()` would get the refresh() for free.

Reply:
Let's keep this in here for now so that `Environ()` doesn't have any unexpected side effects.  When we revisit this to implicitly save on value change we can look to see if we want an alternate approach.